corehq/apps/accounting/forms.py

```diff
@@ -566,7 +566,7 @@ def clean(self):
             and start_date > self.cleaned_data['end_date']):
             raise ValidationError("End date must be after start date.")
 
-        if self.cleaned_data['end_date'] <= datetime.date.today():
+        if self.cleaned_data['end_date'] and self.cleaned_data['end_date'] <= datetime.date.today():
```

Comment:
Dates and times in billing always correspond to the server timezone.  ```end_date``` is submitted by the user.